stdlib/2and3/tarfile.pyi

```diff
@@ -77,7 +77,7 @@ class TarFile(Iterable[TarInfo]):
     mode: str
     fileobj: Optional[IO[bytes]]
     format: Optional[int]
-    tarinfo: Optional[TarInfo]
+    tarinfo: Optional[Type[TarInfo]]
```

Comment:
Could you remove `Optional` here as well? While the constructor argument can be `None`, this field is initialized to `TarInfo`.